cognite/client/data_classes/capabilities.py

```diff
@@ -255,6 +398,12 @@ class AssetRootIDScope(Capability.Scope):
     _scope_name = "assetRootIdScope"
     root_ids: list[int]
 
+    def __post_init__(self) -> None:
+        object.__setattr__(self, "root_ids", [int(i) for i in self.root_ids])
```

Comment:
Not saying we should not do this, but it will cause some pain for the template repo as there  you store a configuration with the id as an external id, and then load it into groups, and then converts all the external ids to ids. Main problem being that you cannot store access configs as internal ids.

Reply:
I don't understand this, could you elaborate? Do you pretend the root_ids are external ids, then convert later?

Reply:
Yes, although I hate that solution, but I can imagen a typical way to store configs is
```
capabilities:
  - threedAcl:
      actions:
        - READ
      scope:
        datasetScope: {
          ids: ['infield_fornebu_location_source_data_dataset']
        }
  - assetsAcl:
      actions:
        - READ
      scope:
        datasetScope: {
          ids: ['infield_fornebu_location_source_data_dataset']
        }
```

Reply:
Oh I see. The problem now is that in order to properly compare capabilities, the ids returned by the API are still strings, even though the spec says integers, and so a user would likely input integers, and then he comparison would fail...